deploy/ocpbugs-15043/script.sh

```diff
@@ -0,0 +1,136 @@
+#!/bin/bash
+
+set -euo pipefail
+
+CHECKED_NAMESPACES=(
+    "openshift-sre-pruning"
+    "openshift-monitoring"
+)
+
+LAST_SL_CONFIGMAP="ocpbugs-15043-last-sl-sent"
+LAST_RUN_ANNOTATION="lasttimestamp"
+DELAY_BETWEEN_SEND_SECONDS="86400"
+
+NODES=()
+
+MACHINE_API_NAMESPACE="openshift-machine-api"
+
+for current_ns in "${CHECKED_NAMESPACES[@]}"; do
+    readarray -t failedPods < <(oc -n "$current_ns" get pods -owide | grep "CreateContainerError" | awk '{print $1}')
+    for pod in "${failedPods[@]}"; do
+        echo "Found pod $pod failing."
+        podDescribe="$(oc -n "$current_ns" describe pod "$pod")"
+
+        if ! echo "$podDescribe" | grep "error loading seccomp filter into kernel" >/dev/null; then
+            echo -e "\t$pod is failing for another reason. Ignoring"
+            continue
+        fi
+
+        node="$(echo "$podDescribe" | grep Node: | awk '{print $2}' | cut -d "/" -f 1)"
+        echo -e "\t$pod on $node having seccomp issues"
+        if echo "${NODES[@]}" | grep "$node" >/dev/null; then
+            continue
+        fi
+        NODES+=("$node")
+    done
+done
+
+if [ ${#NODES[@]} -eq 0 ]; then
+    echo "No affected nodes."
+    exit 0
+fi
+
+# print affected nodes and 
+
+for NODE in "${NODES[@]}"; do
+    nodeJson="$(oc get node "$NODE" -ojson)"
+
+    # make sure we only restart infra nodes
+    if ! jq -e '.metadata.labels."node-role.kubernetes.io/infra"' <<< "$nodeJson" > /dev/null ; then
+      echo "Found node '$node', however control-plane and worker nodes must be restarted manually"
+      continue
+    fi
+
+    echo "Found node '$node'. Checking associated machine cr"
+
+    MACHINE="$(jq -r '.metadata.annotations."machine.openshift.io/machine"' <<< "$nodeJson" | cut -d '/' -f2)"
+    if [[ $MACHINE = "" ]]; then
+      echo "Couldn't find Machine for node $node. Skipping"
+      continue
+    fi
+
+    echo "Found machine '$MACHINE' for node '$node'. Removing it..."
+    oc -n "$MACHINE_API_NAMESPACE" delete machine "$MACHINE" --wait
```

Comment:
should this not delete a machine already in a delete state?

Reply:
What do you mean exactly? This command will wait until the targeted machine has been completely deleted.